File: client/registration/src/app/shared/domain/Job.ts
import {User} from "./User";

export class Job{
    id: number;
    jobName: string;
    jobState: string;
    jobDescribe: string;
    jobSalary: string;
    user: User;
    address: string;
    contact: string;
    updateTime: string;
    constructor(arg?:{
        id: number,
        jobName: string,
        jobState: string,
        jobDescribe: string,
        jobSalary: string,
        user: User,
        address: string,
        contact: string,
        updateTime: string,
    }){
        if(!arg){
            return;
        }

        this.id = arg.id;
        this.jobDescribe = arg.jobDescribe;
        this.jobName = arg.jobName;
        this.jobSalary = arg.jobSalary;
        this.jobState = arg.jobState;
        this.user = arg.user;
        this.address = arg.address;
        this.updateTime = arg.updateTime;
        this.contact = arg.contact;
    }

}
// private Long id;
// private String jobName;
// private String jobState;
// // 更新时间
// private String updateTime;
// private String jobDescribe;
// private String jobSalary;
// @ManyToOne
// private User user;

File: server/src/main/java/com/shnu/part/domain/Job.java
package com.shnu.part.domain;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class Job {
    @Id
    @GeneratedValue
    private Long id;
    private String jobName;
    private String jobState;
    // 更新时间
    private String updateTime;
    private String jobDescribe;
    private String jobSalary;
    @ManyToOne
    private User user;
    private String contact;
    private String address;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getJobName() {
        return jobName;
    }

    public void setJobName(String jobName) {
        this.jobName = jobName;
    }

    public String getJobState() {
        return jobState;
    }

    public void setJobState(String jobState) {
        this.jobState = jobState;
    }

    public String getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(String updateTime) {
        this.updateTime = updateTime;
    }

    public String getJobDescribe() {
        return jobDescribe;
    }

    public void setJobDescribe(String jobDescribe) {
        this.jobDescribe = jobDescribe;
    }

    public String getJobSalary() {
        return jobSalary;
    }

    public void setJobSalary(String jobSalary) {
        this.jobSalary = jobSalary;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}
